Add best practices for writing Prometheus Alerts (#167)
projectsyn/documentation: docs/modules/ROOT/pages/explanations/commodore-components/alerts.adoc

But keep in mind that if all alerts are critical none of them are.

* *Make alerts tunable*
+
You most likely won't be able to write a perfect alert out of the box.
It will either be too noisy, not sensitive enough, or in some other way not relevant for the user.
With that in mind, give the user a way to tune your alert.
+
At the very least provide ways to selectively enable or disable individual alerts.
It's considered best practice to let the user overwrite all of the alert specification if they wish.
However, it's a good idea to also provide some more convenient parameters to tune configuration that often need to be adapted such as alert labels or alert specific parameters like a list of relevant namespaces.
+
Try to imagine what a user might need to change and make tuning it as easy as possible.

* *Provide a runbook*
+
You should always provide a link to a runbook in an annotation `runbook_url`.
See the section below on writing good runbooks.

Following these guidelines, you should get a usable alert.
There are still some pitfalls when writing Prometheus alerts, but there are also many guides to help you write them.

> You can look at https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/[the official documenation] or check out how https://blog.cloudflare.com/monitoring-our-monitoring/[Cloudscale writes alert rules].

Reviewer: ```suggestion
You can look at https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/[the official documentation] or check out how https://blog.cloudflare.com/monitoring-our-monitoring/[Cloudscale writes alert rules].
```

[WARNING]
====
When installing third party software there are often upstream alerts.
It's a good idea to reuse these alerts, but the best practices still apply.

Don't blindly include all upstream alerts.
Check if they're actionable, add labels, make them tunable, and provide a runbook, even if you didn't write the alert yourself.
====

== Writing Runbooks

Every alert rule should have a runbook.
The runbook is the first place a user looks to get information on the alert and how to debug it.

It should tell the reader:

* *What does this alert mean?*
+
Tell the reader why they got the alert.
What exactly doesn't work as it should?
Maybe also tell the user how the alert was measured and if there might be false positive.
* *What's the impact?*
+
Who and what's effected?
How fast should the reader react?
The alert labels should already give an impression how critical the alert is, but try to be more explicit in the runbook.
* *How do I diagnose this?*
+
Provide some input on how to debug this.
Where might the reader get the relevant events or logs?

> How can he narrow down the possible root causes.

Reviewer 2: ```suggestion
How to narrow down the possible root causes?
```

Gender-neutral.
